### ISSUE TITLE
Uncaught TypeError: undefined is not a function

### ISSUE THREAD

**Issue by arunjit** [joefitzgerald/go-plus#116]
This happens when I try to open a .go file, either from the sidebar or Cmd+P.
The file does open when I click the name on the sidebar, but no tab is created (ie, the editor area shows the code and is editable and save works).

[There is already an open issue for the tab bar]

Uncaught TypeError: undefined is not a function

**Atom Version**: 0.162.0
**System**: Mac OS X 10.10.1
**Thrown From**: [go-plus](https://github.com/joefitzgerald/go-plus) package, v3.0.11

### Steps To Reproduce

1. Open a go file

### Stack Trace

```
At /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/node_modules/pathwatcher/node_modules/q/q.js:126

TypeError: undefined is not a function
  at Gocode.module.exports.Gocode.registerProvider (/Users/arunjit/.atom/packages/go-plus/lib/gocode.coffee:38:19)
  at Gocode.registerProvider (/Users/arunjit/.atom/packages/go-plus/lib/gocode.coffee:1:1)
  at /Users/arunjit/.atom/packages/go-plus/lib/gocode.coffee:33:78
  at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/src/workspace.js:270:16
  at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/src/workspace.js:327:18
  at Emitter.module.exports.Emitter.emit (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:82:11)
  at PaneContainer.module.exports.PaneContainer.addedPaneItem (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/src/pane-container.js:379:27)
  at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/src/pane-container.js:366:26
  at Emitter.module.exports.Emitter.emit (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:82:11)
  at Pane.module.exports.Pane.addItem (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/src/pane.js:334:20)
  at Pane.module.exports.Pane.activateItem (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/src/pane.js:313:14)
  at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/src/workspace.js:494:16
  at _fulfilled (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/node_modules/pathwatcher/node_modules/q/q.js:787:54)
  at self.promiseDispatch.done (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/node_modules/pathwatcher/node_modules/q/q.js:816:30)
  at Promise.promise.promiseDispatch (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/node_modules/pathwatcher/node_modules/q/q.js:749:13)
  at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/node_modules/pathwatcher/node_modules/q/q.js:509:49
  at flush (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app/node_modules/pathwatcher/node_modules/q/q.js:108:17)
  at process._tickCallback (node.js:378:11)

```

/cc @atom/core

**Comment by joefitzgerald** [comment]
Can you please update Atom, `autocomplete-plus`, and `go-plus` to the latest versions and let me know if you're still experiencing this issue?

**Comment by drewblas** [comment]
I too have this issue

1. Open a go file
2. Cmd+Shift+P and select "Gocover"

**Atom Version**: 0.177.0
**System**: Mac OS X 10.10.2
**Thrown From**: [go-plus](https://github.com/joefitzgerald/go-plus) package, v3.1.3

### Stack Trace

Uncaught TypeError: undefined is not a function

```
At /Users/drew/.atom/packages/go-plus/lib/gocover.coffee:156

TypeError: undefined is not a function
  at /Users/drew/.atom/packages/go-plus/lib/gocover.coffee:156:7
  at exit (/Users/drew/.atom/packages/go-plus/lib/executor.coffee:64:7)
  at BufferedProcess.triggerExitCallback (/Applications/Atom.app/Contents/Resources/app/src/buffered-process.js:63:47)
  at /Applications/Atom.app/Contents/Resources/app/src/buffered-process.js:77:18
  at Socket.<anonymous> (/Applications/Atom.app/Contents/Resources/app/src/buffered-process.js:150:18)
  at Socket.emit (events.js:138:20)
  at Pipe.close (net.js:477:12)

```

### Commands

```
     -2:20.3 command-palette:toggle (atom-text-editor.editor.vim-mode.autocomplete-active.command-mode)
  4x -2:18.0 core:backspace (atom-text-editor.editor.mini)
     -2:09.6 core:confirm (atom-text-editor.editor.mini)
     -2:09.6 golang:gocover (atom-text-editor.editor.vim-mode.autocomplete-active.command-mode)
     -0:16.9 command-palette:toggle (atom-text-editor.editor.vim-mode.autocomplete-active.command-mode)
     -0:14.7 core:confirm (atom-text-editor.editor.mini)
     -0:14.7 golang:cleargocover (atom-text-editor.editor.vim-mode.autocomplete-active.command-mode)
     -0:13.1 command-palette:toggle (atom-text-editor.editor.vim-mode.autocomplete-active.command-mode)
     -0:11.7 editor:consolidate-selections (atom-text-editor.editor.mini)
     -0:11.7 core:cancel (atom-text-editor.editor.mini)
     -0:11.5 vim-mode:reset-command-mode (atom-text-editor.editor.vim-mode.autocomplete-active.command-mode)
  2x -0:11.2 core:close (atom-text-editor.editor.vim-mode.autocomplete-active.command-mode)
     -0:06.1 command-palette:toggle (atom-text-editor.editor.vim-mode.command-mode.is-focused)
  2x -0:03.8 core:backspace (atom-text-editor.editor.mini)
     -0:02.6 core:confirm (atom-text-editor.editor.mini)
     -0:02.5 golang:gocover (atom-text-editor.editor.vim-mode.command-mode.is-focused)
```

### Config

```json
{
  "core": {
    "ignoredNames": [
      ".git",
      ".svn",
      ".DS_Store"
    ],
    "themes": [
      "atom-dark-ui",
      "twilight"
    ],
    "disabledPackages": [
      "metrics"
    ]
  },
  "go-plus": {
    "goExecutablePath": "/usr/local/bin/go",
    "gofmtPath": "/usr/local/bin/gofmt",
    "vetArgs": "-composites=false -shadow"
  }
}
```

### Installed Packages

```coffee
# User
atom-jinja2, v0.5.0
atom-runner, v2.2.0
autocomplete-plus, v2.1.0
color-picker, v1.4.4
command-logger, v0.22.0
dash, v1.0.2
gist-it, v0.6.4
go-oracle, v0.2.0
go-plus, v3.1.3
jshint, v1.2.0
language-elixir, v0.3.3
language-ini, v1.3.0
linter, v0.11.1
linter-elixirc, v0.2.1
remote-edit, v1.6.3
smarter-delete-line, v1.0.0
twilight, v0.3.0
vim-mode, v0.30.0
wrap-guide, v0.31.0

# Dev
No dev packages
```

/cc @atom/core

**Comment by joefitzgerald** [comment]
This is fixed in 3.2.0.

**Comment by arunjit** [comment]
Fixed in the update, thanks!
